server.go

```diff
@@ -686,7 +686,10 @@ func (sp *serverPeer) OnVersion(_ *peer.Peer, msg *wire.MsgVersion) {
 	remoteAddr := sp.NA()
 	addrManager := sp.server.addrManager
 	if !cfg.SimNet && !cfg.RegNet && !isInbound {
-		addrManager.SetServices(remoteAddr, msg.Services)
+		err := addrManager.SetServices(remoteAddr, msg.Services)
+		if err != nil {
+			srvrLog.Debugf("Setting services for address failed: %v", err)
```

Comment:
What's the reason for preferring `Debugf` here instead of `Errorf` here and the instances below? Should we not be returning on these errors?

Reply:
`Errorf` could be more appropriate since the address should be in the address manager at this point, updated.

As far as returning and disconnecting, it wasn't returning prior to this change if the address was not in the address manager so I left it to follow the same path as a part of this refactor.

Reply:
It appears that permanent peers were/are not added to the address manager so I reverted this back to `Debugf`.  There might be some other category of addresses also not added to the address manager but I'm not entirely sure. Although this does need to be accounted for so this branch is not hit, my intention for this PR was non-functional changes.

Reply:
Noted, and yes I think the peer should be disconnected from if `SetServices` or `Good` fails.